txpipe/noise_maps.py

```diff
@@ -184,17 +185,31 @@ def run(self):
                 for i in range(clustering_realizations):
                     # We have computed the first half already,
                     # and we have the total map from an earlier stage
-                    half1 = ngal_split[:, b, i, 0]
-                    half2 = ngal_split[:, b, i, 1]
-
+                    #half1 = ngal_split[:, b, i, 0]
+                    #half2 = ngal_split[:, b, i, 1]
+
+                    half1 = np.zeros(npix)
+                    half2 = np.zeros_like(half1)
+
+                    if self.config['mask_in_weights']:
+                        half1 = ngal_split[:, b, i, 0]
+                        half2 =	ngal_split[:, b, i, 1]
+                    else:
+                        half1 =	(ngal_split[:, b, i, 0])/mask[pixels]
+                        half2 = (ngal_split[:, b, i, 1])/mask[pixels]
+                    
                     # Convert to overdensity.  I thought about
                     # using half the mean from the full map to reduce
                     # noise, but thought that might add covariance
                     # to the two maps, and this shouldn't be that noisy
-                    mu1 = np.average(half1, weights=mask[pixels])
-                    mu2 = np.average(half2, weights=mask[pixels])
+                    #mu1 = np.average(half1, weights=mask[pixels])
+                    #mu2 = np.average(half2, weights=mask[pixels])
                     # This will produce some mangled sentinel values
                     # but they will be masked out
+
+                    mu1 = np.mean(half1)
```

Comment:
Is there a reason we don't want the weights here?  I haven't thought this through though!

Reply:
@fjaviersanchez what was the reason for removing the weights here?

Reply:
Oh, sorry! It's because I am assuming that the only weights are due to the mask and that's taken into account in lines 198 and 199. If there are additional weights we would have to use `np.average` with the weights. For the HSC branch, so far, we only have the mask and no weights (we use deprojection instead) so the weights were not relevant. I think that for the general case where you have non-unity weights for all galaxies, you would need to bring that weighted average back.

Reply:
Thanks Javi! OK, I'll revert these changes then.